ctwrap/modules/adiabatic_flame.py

```diff
@@ -94,28 +89,77 @@ def run(name,
     if loglevel > 0:
         f.show_solution()
     msg = '    {0:s}: mixture-averaged flamespeed = {1:7f} m/s'
-    print(msg.format(name, f.u[0]))
+    print(msg.format(name, f.velocity[0]))
 
-    out[name + '<mix>'] = pd.DataFrame(
-        np.vstack([f.grid, f.u, f.V, f.T, f.density, f.X]).T, columns=keys)
+    group = name + "<mix>"
+    out[group] = f
 
     # Solve with multi-component transport properties
     f.transport_model = 'Multi'
     f.solve(loglevel)  # don't use 'auto' on subsequent solves
     if loglevel > 0:
         f.show_solution()
     msg = '    {0:s}: multi-component flamespeed  = {1:7f} m/s'
-    print(msg.format(name, f.u[0]))
+    print(msg.format(name, f.velocity[0]))
 
-    out[name + '<multi>'] = pd.DataFrame(
-        np.vstack([f.grid, f.u, f.V, f.T, f.density, f.X]).T, columns=keys)
+    group = name + "<multi>"
+    out[group] = f
 
     return out
 
 
-###
+def save(data: Dict[str, Any], output: Optional[Dict[str, Any]] = None,
```

Comment:
OK, I believe this needs to be much simpler:
```python
def save(data: Dict[str, ct.FlameBase]):

    for group, flame in data.items():
        flame.write_hdf(group, description)
```
This is one way of doing it, but the point here is that the `save` method needs to be short and sweet.